python/src/GmodPath.py

```diff
@@ -0,0 +1,577 @@
+            self._path = path
+            self._current_index = -1 
+            self._current_node = None
+            self._from_depth = from_depth
+
+            if from_depth is not None:
+                if from_depth < 0 or from_depth > len(self._path.parents):
+                    raise IndexError("from_depth out of range")
+                self._current_index = from_depth - 1
+
+        def __iter__(self):
+            return self
+
+        def __next__(self) -> Tuple[int, GmodNode]:
+            if self._current_index < len(self._path.parents):
+                self._current_index += 1
+                if self._current_index == len(self._path.parents):
+                    self._current_node = self._path.node
+                else:
+                    self._current_node = self._path.parents[self._current_index]
+                return self._current_index, self._current_node
+            else:
+                raise StopIteration
+
+        def reset(self) -> None:
+            self._current_index = self._from_depth - 1 if self._from_depth is not None else -1
+
+    def get_full_path(self) -> GmodPath.Enumerator:
+        return self.Enumerator(self)
+
+    def get_full_path_from(self, from_depth : int) -> GmodPath.Enumerator:
+        if from_depth < 0 or from_depth > len(self.parents):
+            raise IndexError("fromDepth is out of allowed range")
+        return self.Enumerator(self, from_depth)
+    
+    def get_normal_assignment_name(self, node_depth: int) -> Optional[str]:
+        node = self[node_depth]
+        normal_assignment_names = node.metadata.normal_assignment_names
+        
+        if not normal_assignment_names:
+            return None
+
+        for i in range(len(self.parents) - 1, -1, -1):
+            child = self[i]
+            name = normal_assignment_names.get(child.code)
+            if name:
+                return name
+
+        return None
+
+    def get_common_names(self):
+        for depth, node in enumerate(self.get_full_path()):
+            is_target = depth == len(self.parents)
+            if not (node[1].is_leaf_node() or is_target) or not node[1].is_function_node():
+                continue
+
+            name = node[1].metadata.common_name or node[1].metadata.name
+            normal_assignment_names = node[1].metadata.normal_assignment_names
+
+            if normal_assignment_names:
+                assignment = normal_assignment_names.get(self.node.code)
+                if assignment:
+                    name = assignment
+
+                for i in range(len(self.parents) - 1, depth - 1, -1):
+                    assignment = normal_assignment_names.get(self.parents[i].code)
+                    if assignment:
+                        name = assignment
+
+            yield (depth, name)
+
+    def get_verbose_path(self):
+        return "/".join([f"{node[1].code} ({node[1].metadata.name})" for node in self.get_full_path()])
```

Comment:
Her mangler du location. Koden skal vel heller ikke være med i denne? Jeg finner heller ingen referanser til funksjonen. Mangler det test?

Reply:
Metadata.name er heller ikke godt nok. Du må bruke get_common_names for å få riktig navn her. Se LocalId.ToString og verbose delen

Reply:
Eksempel er: C101.Name = "Engine", men 411.1/C101 skal bli Propulsion engine, ikke Propulsion Arrangement/Engine